phpBB/includes/acp/acp_board.php

```diff
@@ -510,6 +510,30 @@ function main($id, $mode)
 			}
 		}
 
+		if ($mode == 'avatar' && $cfg_array['allow_avatar_upload'])
+		{
+			$filesystem = $phpbb_container->get('filesystem');
```

Comment:
Move this to inside the else branch. It's not needed if the avatar path is empty.